python_modules/dagster/dagster/core/definitions/resource_requirement.py

```diff
@@ -197,3 +199,30 @@ def ensure_requirements_satisfied(
             raise DagsterInvalidDefinitionError(
                 f"{requirement.describe_requirement()} was not provided{mode_descriptor}. Please provide a {str(requirement.expected_type)} to key '{requirement.key}', or change the required key to one of the following keys which points to an {str(requirement.expected_type)}: {requirement.keys_of_expected_type(resource_defs)}"
             )
+
+
+def get_and_validate_transitive_resource_dependencies(
+    resource_defs: Dict[str, "ResourceDefinition"],
```

Comment:
Nitpick: prefer Mapping to Dict

Reply:
I think at this point we can verify that it's dict. Agreed that we prefer that we use Mapping when ingesting resource defs, but internally we should transform to dict so that stuff like `merge_dicts` works properly.

Reply:
The point of using `Mapping` instead of `Dict` is that it communicates that the function will not mutate the object that's passed to it, which is a helpful invariant for reasoning about what code does. `merge_dicts` actually accepts `Mapping` objects.

Reply:
hmm I thought I had gotten errors from using `merge_dicts` with mapping in the past, but perhaps that's since been rectified. Will change.